src/k8s-engine/lima.ts

```diff
@@ -1164,4 +1164,58 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
   async setIntegration(linkPath: string, state: boolean): Promise<string | undefined> {
     return await this.unixlikeIntegrations.setIntegration(linkPath, state);
   }
+
+  async configureDockerSocket() {
+    const targetPath = path.join(paths.lima, MACHINE_NAME, 'sock', 'docker');
+    const symlinkPath = '/var/run/docker.sock';
+
+    try {
+      const realPath = await fs.promises.realpath(symlinkPath);
+
+      if (realPath === targetPath) {
+        // already linking /var/run/docker.sock to the socket lima manages
+        return;
+      }
+    } catch (err) {
+      if (err.code !== 'ENOENT') {
+        console.log(`Error trying to get the real path for ${ symlinkPath }: `, err);
+      }
+    }
+    const randomTag = Math.random().toString().substr(0, 8);
+    const tmpScript = path.join(os.tmpdir(), `rd-unlink-docker-commands${ randomTag }.sh`);
+    const logFile = path.join(os.tmpdir(), `rd-unlink-docker-commands-run${ randomTag }.log`);
+
+    await fs.promises.writeFile(tmpScript, `#!/usr/bin/env bash
+exec  &> >(tee ${ logFile })
+set -ex
+
+ln -sf "${ targetPath }" "${ symlinkPath }"
+# chown root:daemon "${ symlinkPath }"
+chmod au+rx "${ symlinkPath }"
+`,
+    { mode: 0o700 });
+    try {
+      const x = await new Promise((resolve, reject) => {
+        sudo.exec(tmpScript, { name: 'Rancher Desktop' }, (error, stdout, stderr) => {
```

Comment:
Is this going to be tied into the combined `sudo` handling once the vmnet changes have been merged?

Reply:
Good point, and I thought of that, although this is a different implementation.

The difference is that the vmnet changes always need to be done (usually on a first run).  These changes don't need to be done if the user never runs moby. So I decided to leave the (future merged code) making two separate calls to `sudo.exec`, and test to verify that it only prompts the user for the password the first time.  If there are two prompts, then this code should be moved into the installation code from the vmnet PR, to keep things simple.

So this PR depends on PR 883.